packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -132,7 +175,12 @@ function UnconnectedNavigatorProvider(
 				}
 
 				return [
-					...prevLocationHistory.slice( 0, -1 ),
+					...prevLocationHistory.slice(
+						prevLocationHistory.length > MAX_HISTORY_LENGTH - 2
```

Comment:
The current check will enforce an effective max length of `MAX_HISTORY_LENGTH - 1`. To enforce a max length of `MAX_HISTORY_LENGTH` we should change it to

```suggestion
						prevLocationHistory.length > MAX_HISTORY_LENGTH - 1
```